tests-invoke

```diff
@@ -242,10 +242,6 @@ class PullTask(object):
             self.start_publishing(opts.publish)
             os.environ["TEST_ATTACHMENTS"] = self.sink.attachments
 
-        head = subprocess.check_output(["git", "rev-parse", "HEAD"], universal_newlines=True).strip()
-        if not self.revision:
-            self.revision = head
```

Comment:
From tests-invoke's perspective, it's not guaranteed to be defined. It does `revision = os.environ.get("TEST_REVISION")`, so it could happen that this env var is simply not set. I agree that it *should* be always set to have a meaningful collision detection and publishing, so if we use a sink it's mandatory.

So if publishing is on (`$TEST_PUBLISH` or `--publish`), please change the `os.environ.get()` to a `os.environ[]` and give a meaningful error message on `KeyError` -- i. e. let's enforce/assert that it's always set.

Reply:
Put there check as you suggested. And also noticed that we read revision from env variable and then set this variable back in, so I dropped it as well.